### PR TITLE
[octavia-ingress-controller] Support internal/external Ingress

### PR DESCRIPTION
**Pull request by lingxiankong**
**What this PR does / why we need it**:
Allow the k8s user choose to create internal or external Ingress by speficying an annotation `octavia.ingress.kubernetes.io/internal`, if it's true, the load balancer created in Octavia won't have floating IP associated. The default value is true.

**Which issue this PR fixes**:

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/ingress/controller/controller.go [hunk 1]
```diff
@@ -66,6 +67,12 @@ const (
 	// LabelNodeRoleMaster specifies that a node is a master
 	// It's copied over to kubeadm until it's merged in core: https://github.com/kubernetes/kubernetes/pull/39112
 	LabelNodeRoleMaster = "node-role.kubernetes.io/master"
+
+	// IngressAnnotationInternal is the annotation used on the Ingress
+	// to indicate that we want an internal loadbalancer service so that octavia-ingress-controller won't associate
+	// floating ip to the load balancer VIP.
+	// Default to true.
+	IngressAnnotationInternal = "octavia.ingress.kubernetes.io/internal"
```
Comment (adisky) [inline pkg/ingress/controller/controller.go]: @lingxiankong it needs a doc update?

Comment (jichenjc) [reply pkg/ingress/controller/controller.go]: +1 for a document change

**Comment by jichenjc** [comment]
@lingxiankong 
just from study purpose, any document on internal/external ingress link?Thanks

**Comment by jichenjc** [comment]
ok, I got it 
https://kubernetes.io/docs/concepts/services-networking/service/#internal-load-balancer

**Comment by jichenjc** [comment]
service.beta.kubernetes.io/openstack-internal-load-balancer: "true"

looks like this is different to proposed name...

File: pkg/ingress/controller/controller.go (hunk 1 above)
Comment (lingxiankong) [reply pkg/ingress/controller/controller.go]: Hi, @adisky @jichenjc, i was planning to update the whole octavia-ingress-controller doc in a separate PR after this PR and #502 merged. Stay tuned :-)

**Comment by lingxiankong** [comment]
> service.beta.kubernetes.io/openstack-internal-load-balancer: "true"
> 
> looks like this is different to proposed name...

The one you mentioned is for LoadBalancer type service, this one is for Ingress. The two annotations are for the same purpose but towards different kubernetes resources.

File: pkg/ingress/controller/controller.go (hunk 1 above)
Comment (adisky) [reply pkg/ingress/controller/controller.go]: @lingxiankong thanks!

**Pull request merged**